linux-documentation/secure_launch_overview.rst

```diff
@@ -0,0 +1,110 @@
+
+- Secure Launch Overview (secure_launch_overview.rst)
+  - TrenchBoot Project
+  - Secure Launch Goals
+  - Functional Overview
+
+
+----
+__From last Patch Set:__
+
+
+=============
+Secure Launch
+=============
+
+Overview
+========
+
+Prior to the start of the TrenchBoot project, the only active Open
+Source project supporting dynamic launch was Intel's tboot project to
+support their implementation of dynamic launch known as Intel Trusted
+eXecution Technology (TXT). The approach taken by tboot was to provide
+an exokernel that could handle the launch protocol implemented by
+Intel's special loader, the SINIT Authenticated Code Module (ACM [3]_)
+and remained in memory to manage the SMX CPU mode that a dynamic launch
+would put a system. While it is not precluded from being used for doing
```

Comment:
This sentence is too complex and hard to understand. Does it mean tboot remains in memory to manage SMX CPU mode?

Should it end with "dynamic launch would put a system in"?

Reply:
This is something I have tried to explain before, most people don't realize but tboot really is a persistent [exokernel](https://en.wikipedia.org/wiki/Exokernel) with the only hardware operations that it manages is power management to manage exit and re-entry (across sleep states) of SMX. This is because it is how Safer Mode Execution was envisioned to work but the reality is that x86 sleep is far more complex and very difficult to determine with any good level assurance that the resumed system is really the same system.

Reply:
Hmm, if that was complex then what will happen with S0ix? Maybe in other place we should explain deeper relation between sleep modes and late launch?